keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepository.kt

```diff
@@ -108,6 +110,22 @@ interface ResourceRepository : PeriodicallyCheckedRepository<Resource<out Resour
    */
   fun deleteByApplication(application: String): Int
 
+  /**
+   * Retrieves the history of persisted events for [application].
+   *
+   * @param application the name of the application.
+   * @param limit the maximum number of events to return.
+   */
+  fun applicationEventHistory(application: String, limit: Int = DEFAULT_MAX_EVENTS): List<ApplicationEvent>
+
+  /**
+   * Retrieves the history of persisted events for [application].
+   *
+   * @param application the name of the application.
+   * @param downTo the time of the oldest event to return.
+   */
+  fun applicationEventHistory(application: String, downTo: Instant): List<ApplicationEvent>
```

Comment:
My instinct is to add a `limit` parameter to all API calls that return a list of objects, just like there is for the previous call.

Reply:
I decided not to add this now because the whole point of this variant of the method was not to be bound by an arbitrary limit and instead make sure we look into the application event history in the exact same time window as the resource events we're returning. Hopefully, since those _are_ subject to a limit, the number of application events returned will not be a problem.